demos/object_detection_demo_ssd_async/detection_model_ssd.cpp

```diff
@@ -0,0 +1,168 @@
+                inputInfoItem.second->getInputData()->setLayout(Layout::NHWC);
+            }
+            else {
+                inputInfoItem.second->getInputData()->setLayout(Layout::NCHW);
+            }
+            const TensorDesc& inputDesc = inputInfoItem.second->getTensorDesc();
+            netInputHeight = getTensorHeight(inputDesc);
+            netInputWidth = getTensorWidth(inputDesc);
+        }
+        else if (inputInfoItem.second->getTensorDesc().getDims().size() == 2) {  // 2nd input contains image info
+            inputsNames.resize(2);
+            inputsNames[2] = inputInfoItem.first;
+            inputInfoItem.second->setPrecision(Precision::FP32);
+        }
+        else {
+            throw std::logic_error("Unsupported " +
+                std::to_string(inputInfoItem.second->getTensorDesc().getDims().size()) + "D "
+                "input layer '" + inputInfoItem.first + "'. "
+                "Only 2D and 4D input layers are supported");
+        }
+    }
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    slog::info << "Checking that the outputs are as the demo expects" << slog::endl;
+    OutputsDataMap outputInfo(cnnNetwork.getOutputsInfo());
+    if (outputInfo.size() != 1) {
+        throw std::logic_error("This demo accepts networks having only one output");
+    }
+    DataPtr& output = outputInfo.begin()->second;
+    outputsNames.push_back(outputInfo.begin()->first);
+
+    int num_classes = 0;
+
+/*    if (auto ngraphFunction = cnnNetwork.getFunction()) {
+        for (const auto op : ngraphFunction->get_ops()) {
+            if (op->get_friendly_name() == outputsNames[0]) {
+                auto detOutput = std::dynamic_pointer_cast<ngraph::op::DetectionOutput>(op);
+                if (!detOutput) {
+                    THROW_IE_EXCEPTION << "Object Detection network output layer(" + op->get_friendly_name() +
+                        ") should be DetectionOutput, but was " + op->get_type_info().name;
+                }
+
+                num_classes = detOutput->get_attrs().num_classes;
+                break;
+            }
+        }
+    }
+    else {
+        throw std::logic_error("This demo requires IR version no older than 10");
+    }*/
+    if (labels.size()) {
+        if (static_cast<int>(labels.size()) == (num_classes - 1)) {  // if network assumes default "background" class, having no label
+            labels.insert(labels.begin(), "fake");
+        }
+        else if (static_cast<int>(labels.size()) != num_classes) {
+            throw std::logic_error("The number of labels is different from numbers of model classes");
+        }
+    }
+
+    const SizeVector outputDims = output->getTensorDesc().getDims();
+    maxProposalCount = outputDims[2];
+    objectSize = outputDims[3];
+    if (objectSize != 7) {
+        throw std::logic_error("Output should have 7 as a last dimension");
+    }
+    if (outputDims.size() != 4) {
```

Comment:
You must check `outputDims.size()` before accessing elements.

Reply:
Fixed